add popularity info on each lib's page

pvnr0082t opened the pull request:
PR for #60 
It will show `watch`, `star`, `fork` information on each lib's page.
![cdnjs](https://raw.githubusercontent.com/pvnr0082t/Hello/master/Screenshot%20from%202017-01-08%2015-33-54.png)

For the library without git repository(no popularity info), it will show like this:
![no-repo](https://raw.githubusercontent.com/pvnr0082t/Hello/master/Screenshot%20from%202017-01-09%2020-15-50.png)

PeterDaveHello commented:
1. How do you handle the lib with git repo but not on GitHub?
2. Can we just hide the icons if there is no valid info?

pvnr0082t commented:
>1. How do you handle the lib with git repo but not on GitHub?

If there is a git repo but not on GitHub, the return value `METAS[pathname]` will be undefined. The popularity information will still be `null`.

>2. Can we just hide the icons if there is no valid info?

Of course, I just added line10 and line16 in `templates/library.html`. If the popularity info are null, it won't show the icons. I updated the screenshot.

PeterDaveHello commented:
@pvnr0082t Great, let's ask review from your mates and see if they have suggestions.

pvnr0082t commented:
@cdnjs/intern2 Please help me review this pull request, thank you.

PeterDaveHello commented:
@pvnr0082t make sure they know how to run your code

pvnr0082t commented:
ping @cdnjs/intern2

PeterDaveHello commented on templates/library.html:
```diff
@@ -7,6 +7,13 @@
         <div class="col-md-8">
 
     <h1 style="margin-top: 00px;" id="libraryName">{{library.name}}</h1>
+    {{#watch}}{{#star}}{{#fork}}
```
I think we can just check one of them, since they'll always be there together, right?

pvnr0082t replied:
You're right.

PeterDaveHello commented on webServer.js:
```diff
@@ -308,6 +319,13 @@ function start() {
     }
     var licenses = librarylicensesList(library);
     var assets = libraryAssetsList(library, version);
+    var stargazers_count, forks, subscribers_count = null;
```
What's the reason why `subscribers` need to be `null` but the other two don't?

pvnr0082t replied:
It's my fault. fixed!

PeterDaveHello commented on webServer.js:
```diff
@@ -255,6 +257,15 @@ function start() {
   }
 
+  function libraryMetaInfo(library) {
+    if (library.repository !== undefined) {
```
I'll suggest you to directly drop those non-GitHub repo here since it's a simply match, there is no need to do an additional parse and string handling to waste the resource.

pvnr0082t replied:
Okay, the library without repo info will directly return null now. :+1:

PeterDaveHello commented on webServer.js:
```diff
@@ -63,6 +63,8 @@ function start() {
     // Load libraries into ram
   var LIBRARIES = JSON.parse(fs.readFileSync('public/packages.min.json', 'utf8')).packages;
+    // Load GitHub repositories meta data
+  var METAS = JSON.parse(fs.readFileSync('GitHub.repos.meta.json', 'utf8'));
```
`METAS` -> `GITHUB_METAS`

pvnr0082t commented:
@cdnjs/intern2 I fixed the problems that @PeterDaveHello mentioned. Please help me review it, thank you.

PeterDaveHello commented on webServer.js:
```diff
@@ -308,6 +319,13 @@ function start() {
     var assets = libraryAssetsList(library, version);
+    var stargazers_count = null, forks = null, subscribers_count = null;
+    var metaInfo = libraryMetaInfo(library);
+    if (metaInfo !== null && metaInfo !== undefined) {
```
What's the reason it'll be `undefined` here?

pvnr0082t replied:
When the library with GitHub url but we cannot access it, it'll be `undefined`. 
For example: `radiobox.css` with GitHub url: https://github.com/720kb/radiobox but 404 not found. We don't have popularity information for this kind of libraries.

PeterDaveHello commented on webServer.js:
```diff
@@ -255,6 +257,15 @@ function start() {
   }
 
+  function libraryMetaInfo(library) {
+    if (library.repository === undefined) {
```
You still don't check if it's GitHub here.

pvnr0082t replied:
I use `/github.com/.test(library.repository.url)` to check if it's Github url. :+1:

PeterDaveHello commented on webServer.js:
```diff
@@ -255,6 +257,15 @@ function start() {
     });
   }
 
+  function libraryMetaInfo(library) {
```
The name `libraryMetaInfo`  should contain "GitHub" keyword.

pvnr0082t commented:
@cdnjs/intern2 I have fixed the problems that @PeterDaveHello mentioned. Please help me review it, thank you.

PeterDaveHello commented:
@cdnjs/website @cdnjs/dev does this looks good to you guys?

nolsherry commented:
LGTM

PeterDaveHello commented:
Let's give it a try.
